### DISCUSSION TITLE
Debugging slow `Creating an optimized production build`

### DISCUSSION THREAD

**Discussion by bitttttten** [vercel/next.js#68008] [Help]
### Summary

It takes us 5 minutes on our CI (vercel) to reach `Creating an optimized production build`. 

Are there any guides, tips, or tricks to debug a slow Next.JS app? Our app is multi-language and has around 250 pages. We are pre-rendering around 600 pages and leaving the rest to be dynamic.

To me this doesn't seem right since a bare Next.JS project can be at this step for 10-15s.. so we have most likely set something up incorrectly so I'm looking for some tips.

**Comment by icyJoseph** [comment]
Is this project using TypeScript? Or does it use icon libraries, or perhaps has a ton of heavy dependencies?

**Reply by bitttttten** [reply]
It's not using any icon libraries, all our icons are in a turborepo package (packages/icons). Indeed it's using TypeScript, and no heavy dependencie:

```json
{
  "dependencies": {
    "@sentry/nextjs": "7.114.0",
    "framer-motion": "10.18.0",
    "nanoid": "5.0.6",
    "next": "14.2.3",
    "next-intl": "3.13.0",
    "react": "^18.2.0",
    "react-color": "2.19.3",
    "react-cookie": "^7.0.0",
    "react-countdown": "2.3.5",
    "react-dom": "^18.2.0",
    "react-hot-toast": "2.4.1",
    "react-multi-carousel": "2.8.5",
    "react-photo-album": "2.4.0",
    "react-popper": "^2.3.0",
    "tinycolor2": "1.6.0",
    "ts-deepmerge": "^7.0.0",
    "ua-parser-js": "1.0.37"
  }
}
```

We are using some local packages as we are in a monorepo. I will investigate those.

**Reply by bitttttten** [reply]
No icon libraries or heavy dependencies. We have playwright installed, although this is in the "tests" package, so not within our nextjs app.

**Reply by bitttttten** [reply]
Is there perhaps some debug flags or verbose flags, so we can see where the work is being done and then see where nextjs is getting stuck?
